### PR TITLE
[ci] Use consistent style in azure-pipelines.yml

### PR DESCRIPTION
**Pull request by mcy**
These changes include:
- Consistent quotation (single quotes in YAML, only for multi-word
  strings; double-quotes in bash around all variable references).
- Consistent use of [[ instead of [ for testing conditions.
- Consistent indentation within bash.
- Consistent file paths (e.g., when calling a script, prefix with ./
  only when strictly necessary).

### CODE REVIEW

**Comment by mcy** [comment]
Also, given the number of *other* shell scripts we have, perhaps we should consider adding something like e.g. https://github.com/koalaman/shellcheck to our CI lint?

**Comment by imphil** [comment]
Can you motivate your choice of YAML formatting? Multi-word string don't need to be quoted. And if quotes are desired, why not use " for bash and YAML?

**Comment by mcy** [comment]
The exact choice isn't motivated, I just made *an* arbitrary choice, because stuff was all over the place.

I couldn't remember the rules for whether multi-word strings needed to be quoted, but I could unquote those, too.

**Comment by imphil** [comment]
Quotes in YAML are needed if special characters are included. Space isn't a special character, but the set of special characters is rather strange. So it's easiest to remember to always use quotes if non-word characters are included. If you use quotes, use the same type for consistency for both bash and YAML, I don't see a need to distinguish the two.

**Comment by imphil** [comment]
@mcy do you want to follow up on the review comments? Otherwise, please close this PR.

**Comment by mcy** [comment]
Yep, I went ahead and did that. It turns out that all of the yaml strings in that file are safe to not have quotes around, except for one at the very start. I used "" for that one.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -105,130 +105,132 @@ jobs:
...
       #   the GitHub checks API. However, warnings don't return a non-zero
       #   error code and don't fail the build step.
-      ./util/lint_commits.py \
-          --no-merges \
-          --error-msg-prefix='##vso[task.logissue type=error]' \
-          --warning-msg-prefix='##vso[task.logissue type=error]' \
-          $commit_range
+      util/lint_commits.py \
+        --no-merges \
+        --error-msg-prefix="##vso[task.logissue type=error] \
```
Comment (imphil) [inline azure-pipelines.yml]:
```suggestion
        --error-msg-prefix="##vso[task.logissue type=error]" \
```

Comment (mcy) [reply azure-pipelines.yml]: Done.

**Pull request merged**